src/form/src/Form.tsx

```diff
@@ -90,38 +90,41 @@ export default defineComponent({
       validateCallback?: FormValidateCallback,
       shouldRuleBeApplied: ShouldRuleBeApplied = () => true
     ): Promise<void> {
-      await new Promise<void>((resolve, reject) => {
-        const formItemValidationPromises: Array<
-        Promise<{
-          valid: boolean
-          errors?: ValidateError[]
-        }>
-        > = []
-        for (const key of keysOf(formItems)) {
-          const formItemInstances = formItems[key]
-          for (const formItemInstance of formItemInstances) {
-            if (formItemInstance.path) {
-              formItemValidationPromises.push(
-                formItemInstance.internalValidate(null, shouldRuleBeApplied)
-              )
-            }
-          }
+      function resolveCallback (
+        errors?: ValidateError[][]
+      ): Promise<void> | void {
+        if (validateCallback) {
+          validateCallback(errors)
+        } else {
+          return errors ? Promise.reject(errors) : Promise.resolve()
         }
-        void Promise.all(formItemValidationPromises).then((results) => {
-          if (results.some((result) => !result.valid)) {
-            const errors = results
-              .filter((result) => result.errors)
-              .map((result) => result.errors)
-            if (validateCallback) {
-              validateCallback(errors as ValidateError[][])
-            }
-            reject(errors)
```

Comment:
Why not just use an else here?

Reply:
else {
  reject(errors)
}

Reply:
From a functional standpoint, simply adding an else statement would suffice. The reason for this modification is to prevent the program from retaining a Promise instance that may never reject or resolve, while also centralizing and improving the readability of the logic for determining whether to use a Promise approach. Therefore, I extract an internal function to decide whether to return `Promise`.